### PR TITLE
Feature/implement form builder

### PR DESCRIPTION
**Pull request by saralabafi**

### CODE REVIEW

File: app/globalHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            <span className="flex w-8 h-8 bg-blue-100 rounded-full items-center justify-center text-sm px-1 mt-1 align-middle content-center">
+              <UserIcon />
+            </span>
+            <div className="block ps-1">
+              <span>معین سپهری</span>
+              <p className="text-[10px]">example.dourtal.com</p>
+            </div>
+          </div>
+          <span className="absolute inset-y-0 end-0 flex items-center p-1 pointer-events-none bg-white rounded-se-md rounded-ee -md">
+            <svg className="h-5 w-5 text-white" fill="#ddd" viewBox="0 0 20 20">
```
Comment (amirkh330) [inline app/globalHeader.tsx]: use as Image component

File: components/Tools/ContentStructureBuilder/SwitchControl/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        onClick={handleClick}>
+        {/* Switch */}
+        <div
+          className={`bg-blue-500 md:w-6 md:h-6 h-5 w-5 rounded-full shadow-md transform duration-300 ease-in-out${
+            toggle ? '' : toggleClass
+          }`}></div>
+        {/* Tick */}
+        {toggle && (
+          <></>
+          // <svg
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SwitchControl/Switch.tsx]: remove useless comment

File: components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import Switch from './Switch'
+import React, { useState } from 'react'
+
+interface SwitchControlProps {
+  label?: string
+  isCheckedControl: boolean
+  onChange: (checked: boolean) => void
+  disabled: boolean
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx]: is it required?

File: components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        className="flex flex-col p-4 py-2 px-6 bg-slate-100 border border-slate-300 border-dashed rounded-md ">
+        <div className="flex items-center mb-2">
+          <input
+            type="checkbox"
+            id="displayCheckbox"
+            checked={isCheckbox}
+            onChange={handleDisplayChange}
+          />
+          <label className="mr-2" htmlFor="displayCheckbox">
+            نمایش چک باکس
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx]: don`t use directly text

File: components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+          ) : (
+            <div className="relative inline-flex flex-shrink-0 h-6 w-12 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none focus-visible:ring focus-visible:ring-primary">
+              <Switch
+                type="checkbox"
+                className="absolute opacity-0 w-0 h-0"
+                id="toggleSwitch"
+                disabled={disabled}
+                onChange={(checked: boolean) => {
+                  // Handle the toggle change logic
+                  console.log('Toggle changed:', checked)
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SwitchControl/SwitchControl.tsx]: remove console.log

File: public/images/globalHeader/userFill.svg [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 20 20" fill="none">
+  <path d="M10 1.66667C7.70621 1.66667 5.83333 3.53955 5.83333 5.83333C5.83333 8.12712 7.70621 10 10 10C12.2938 10 14.1667 8.12712 14.1667 5.83333C14.1667 3.53955 12.2938 1.66667 10 1.66667ZM5.20833 11.6667C4.18021 11.6667 3.33333 12.5135 3.33333 13.5417V14.0413C3.33333 15.2667 4.1103 16.3658 5.29541 17.1216C6.48052 17.8773 8.10202 18.3333 10 18.3333C11.898 18.3333 13.5195 17.8773 14.7046 17.1216C15.8897 16.3658 16.6667 15.2667 16.6667 14.0413V13.5417C16.6667 12.5135 15.8198 11.6667 14.7917 11.6667H5.20833Z" fill="#3B82F6"/>
+</svg>
```
Comment (amirkh330) [inline public/images/globalHeader/userFill.svg:3]: use currentColor for fill

File: components/Tools/ContentStructureBuilder/SingleSelection/SingleSelection.types.tsx [hunk 7]
```diff
@@ -0,0 +1,11 @@
+export interface ISingleSelection {
+  title: string
+  helpText?: string
+  layoutType?: 'line' | 'list'
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SingleSelection/SingleSelection.types.tsx]: use row | column

File: components/Tools/ContentStructureBuilder/SingleSelection/SingleSelection.tsx [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        <p className="text-red-500 text-[10px] italic pb-2">{errorMassage}</p>
+      ) : null}
+      <div className={`flex ${layout}`}>
+        {options.map((option, i) => (
+          <div className="flex items-center mb-2 me-4">
+            <input
+              id={`${i}`}
+              type="radio"
+              name={title}
+              value=""
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/SingleSelection/SingleSelection.tsx]: don`t need set value?

File: components/Tools/ContentStructureBuilder/NumericControl/NumericControl.type.tsx [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  placeholder?: string
+  helpText?: string
+  numberType?: 'int' | 'float'
+  required?: boolean
+  errorMassage?: string
+  defaultValue?: string
+  searchable?: 'deactive' | 'keyWord'
+  hide?: boolean
+  displayTitle?: boolean
+  onChange?: () => void | undefined
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/NumericControl/NumericControl.type.tsx]: When you use question mark (?), you don`t need to write | undefined

Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/NumericControl/NumericControl.type.tsx]: onChange don`t send any event?

File: components/Tools/ContentStructureBuilder/NumericControl/NumericControl.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        {required ? <span className="text-red-500 me-2">*</span> : null}
+        {displayTitle ? title : null}
+      </p>
+      <input
+        className="border border-slate-200 rounded p-1 mb-2 text-[12px]"
+        placeholder={placeholder}
+        type="number"
+        step={numberType === 'int' ? '1' : 'any'}
+        // type="text"
+        // inputMode="numeric"
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/NumericControl/NumericControl.tsx]: remove  useless comment

File: components/Tools/ContentStructureBuilder/NumericControl/NumericControl.tsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    defaultValue,
+    searchable,
+    hide = false,
+    displayTitle = true,
+    onChange,
+  } = props
+  return (
+    <div className="p-2">
+      <p className="text-[12px] pb-2">
+        {required ? <span className="text-red-500 me-2">*</span> : null}
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/NumericControl/NumericControl.tsx:20]: don`t use direct text
I suggest you use an image or SVG

File: components/Tools/ContentStructureBuilder/TextControl/TextControl.tsx [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+      </label>
+      {multiLine ? (
+        <textarea
+          className={`resize ${inputBorder} block w-full bg-white text-gray-700 border rounded py-3 px-4 mb-3 leading-tight`}
+          required={required}
+          autoComplete={autoComplete}
+          autoFocus={autoFocus}
+          value={value}
+          defaultValue={defaultValue}
+          disabled={disabled ?? false}
```
Comment (amirkh330) [inline components/Tools/ContentStructureBuilder/TextControl/TextControl.tsx]: set default, where you detract object props

**Review by hesamhabibih — APPROVED** [review]

**Pull request merged**